lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -1199,6 +1212,72 @@ TfToken MayaDescriptorToToken(const MVertexBufferDescriptor& descriptor)
 
 } // anonymous namespace
 
+class HdVP2Material::TextureLoadingTask
+{
+public:
+    TextureLoadingTask(
+        HdVP2Material*     parent,
+        HdSceneDelegate*   sceneDelegate,
+        const std::string& path,
+        const VtValue&     fallbackValue)
+        : _parent(parent)
+        , _sceneDelegate(sceneDelegate)
+        , _path(path)
+        , _fallbackValue(fallbackValue)
+    {
+    }
+
+    ~TextureLoadingTask() = default;
+
+    void Load()
+    {
+        if (_started.load()) {
+            return;
+        }
+        _started = true;
+        // Push the texture loading on idle
+        MGlobal::executeTaskOnIdle(
+            [](void* data) {
+                auto* task = static_cast<HdVP2Material::TextureLoadingTask*>(data);
+                task->_LoadOnIdle();
+                // Once it is done, free the memory.
+                delete task;
+            },
+            this);
+    }
+
+    bool Terminate()
+    {
+        _terminated = true;
+        // Return the started state to caller, the caller will delete this object
+        // if this task has not started yet.
+        // We will not be able to delete this object within its method.
+        return !_started.load();
+    }
+
+private:
+    void _LoadOnIdle()
+    {
+        if (_terminated) {
+            return;
+        }
+        bool        isSRGB = false;
+        MFloatArray uvScaleOffset;
+        auto*       texture = _LoadTexture(_path, isSRGB, uvScaleOffset, _fallbackValue);
```

Comment:
_LoadTexture includes the call to see if the texture is already in the texture cache, so this idle loading approach is already working with texture caching and texture re-use.